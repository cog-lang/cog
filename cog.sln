
Microsoft Visual Studio Solution File, Format Version 12.00
# Visual Studio 14
VisualStudioVersion = 14.0.25420.1
MinimumVisualStudioVersion = 10.0.40219.1
Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "cog", "source\cog\cog.vcxproj", "{C59CF5E2-B2D5-46DC-9499-371B3AC26FF3}"
EndProject
Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "cogc-bootstrap", "source\cogc-bootstrap\cogc-bootstrap.vcxproj", "{F3DAF015-5BA0-48E0-A178-E12BF27B5DDB}"
EndProject
Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "cogc", "source\cogc\cogc.vcxproj", "{22FAAF10-49F3-4CCD-B3A8-F3A6ABE2A41E}"
EndProject
Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "runtime", "source\runtime\runtime.vcxproj", "{9E0A7354-29D9-4525-BE90-9C557D8C36AE}"
EndProject
Global
	GlobalSection(SolutionConfigurationPlatforms) = preSolution
		Debug|x64 = Debug|x64
		Debug|x86 = Debug|x86
		Release|x64 = Release|x64
		Release|x86 = Release|x86
	EndGlobalSection
	GlobalSection(ProjectConfigurationPlatforms) = postSolution
		{C59CF5E2-B2D5-46DC-9499-371B3AC26FF3}.Debug|x64.ActiveCfg = Debug|x64
		{C59CF5E2-B2D5-46DC-9499-371B3AC26FF3}.Debug|x64.Build.0 = Debug|x64
		{C59CF5E2-B2D5-46DC-9499-371B3AC26FF3}.Debug|x86.ActiveCfg = Debug|Win32
		{C59CF5E2-B2D5-46DC-9499-371B3AC26FF3}.Debug|x86.Build.0 = Debug|Win32
		{C59CF5E2-B2D5-46DC-9499-371B3AC26FF3}.Release|x64.ActiveCfg = Release|x64
		{C59CF5E2-B2D5-46DC-9499-371B3AC26FF3}.Release|x64.Build.0 = Release|x64
		{C59CF5E2-B2D5-46DC-9499-371B3AC26FF3}.Release|x86.ActiveCfg = Release|Win32
		{C59CF5E2-B2D5-46DC-9499-371B3AC26FF3}.Release|x86.Build.0 = Release|Win32
		{F3DAF015-5BA0-48E0-A178-E12BF27B5DDB}.Debug|x64.ActiveCfg = Debug|x64
		{F3DAF015-5BA0-48E0-A178-E12BF27B5DDB}.Debug|x64.Build.0 = Debug|x64
		{F3DAF015-5BA0-48E0-A178-E12BF27B5DDB}.Debug|x86.ActiveCfg = Debug|Win32
		{F3DAF015-5BA0-48E0-A178-E12BF27B5DDB}.Debug|x86.Build.0 = Debug|Win32
		{F3DAF015-5BA0-48E0-A178-E12BF27B5DDB}.Release|x64.ActiveCfg = Release|x64
		{F3DAF015-5BA0-48E0-A178-E12BF27B5DDB}.Release|x64.Build.0 = Release|x64
		{F3DAF015-5BA0-48E0-A178-E12BF27B5DDB}.Release|x86.ActiveCfg = Release|Win32
		{F3DAF015-5BA0-48E0-A178-E12BF27B5DDB}.Release|x86.Build.0 = Release|Win32
		{22FAAF10-49F3-4CCD-B3A8-F3A6ABE2A41E}.Debug|x64.ActiveCfg = Debug|x64
		{22FAAF10-49F3-4CCD-B3A8-F3A6ABE2A41E}.Debug|x64.Build.0 = Debug|x64
		{22FAAF10-49F3-4CCD-B3A8-F3A6ABE2A41E}.Debug|x86.ActiveCfg = Debug|Win32
		{22FAAF10-49F3-4CCD-B3A8-F3A6ABE2A41E}.Debug|x86.Build.0 = Debug|Win32
		{22FAAF10-49F3-4CCD-B3A8-F3A6ABE2A41E}.Release|x64.ActiveCfg = Release|x64
		{22FAAF10-49F3-4CCD-B3A8-F3A6ABE2A41E}.Release|x64.Build.0 = Release|x64
		{22FAAF10-49F3-4CCD-B3A8-F3A6ABE2A41E}.Release|x86.ActiveCfg = Release|Win32
		{22FAAF10-49F3-4CCD-B3A8-F3A6ABE2A41E}.Release|x86.Build.0 = Release|Win32
		{9E0A7354-29D9-4525-BE90-9C557D8C36AE}.Debug|x64.ActiveCfg = Debug|x64
		{9E0A7354-29D9-4525-BE90-9C557D8C36AE}.Debug|x64.Build.0 = Debug|x64
		{9E0A7354-29D9-4525-BE90-9C557D8C36AE}.Debug|x86.ActiveCfg = Debug|Win32
		{9E0A7354-29D9-4525-BE90-9C557D8C36AE}.Debug|x86.Build.0 = Debug|Win32
		{9E0A7354-29D9-4525-BE90-9C557D8C36AE}.Release|x64.ActiveCfg = Release|x64
		{9E0A7354-29D9-4525-BE90-9C557D8C36AE}.Release|x64.Build.0 = Release|x64
		{9E0A7354-29D9-4525-BE90-9C557D8C36AE}.Release|x86.ActiveCfg = Release|Win32
		{9E0A7354-29D9-4525-BE90-9C557D8C36AE}.Release|x86.Build.0 = Release|Win32
	EndGlobalSection
	GlobalSection(SolutionProperties) = preSolution
		HideSolutionNode = FALSE
	EndGlobalSection
EndGlobal
