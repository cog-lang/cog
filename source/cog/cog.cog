// cog.cog

// This file provides the Cog "standard library"

struct Void {}

struct Bool {}
let true : Bool;
let false : Bool;

struct Char {}

struct Int {}
struct UInt {}

struct Size {}

struct Ptr<T> {}
struct ConstPtr<T> {}

func -(left: Int, right: Int) -> Int;

func - <T>(left: Ptr<T>, right: Ptr<T>) -> Int;

func >(left: Int, right: Int) -> Bool;

