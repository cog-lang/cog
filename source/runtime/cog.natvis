<?xml version="1.0" encoding="utf-8"?>

<AutoVisualizer xmlns="http://schemas.microsoft.com/vstudio/debugger/natvis/2010">

<Type Name="cog::RefPtr&lt;*&gt;">
    <SmartPointer Usage="Minimal">object</SmartPointer>
    <DisplayString Condition="object == 0">null</DisplayString>
    <DisplayString Condition="object != 0">{*object}</DisplayString>
    <Expand>
      <ExpandedItem>object</ExpandedItem>
    </Expand>
</Type>

<Type Name="cog::String">
    <DisplayString>{((char*) (storage.object+1)),s}</DisplayString>
	<StringView>((char*) (storage.object+1)),s</StringView>
</Type>

<Type Name="cog::Array&lt;*&gt;">
  <DisplayString>{{ count={($T1*)storage.object->end - ($T1*)(storage.object + 1)} }}</DisplayString>
  <Expand>
    <Item Name="[count]">($T1*)storage.object->end - ($T1*)(storage.object + 1)</Item>
    <Item Name="[capacity]">($T1*)storage.object->limit - ($T1*)(storage.object + 1)</Item>
    <ArrayItems>
      <Size>($T1*)storage.object->end - ($T1*)(storage.object + 1)</Size>
      <ValuePointer>($T1*)(storage.object + 1)</ValuePointer>
    </ArrayItems>
  </Expand>
</Type>

</AutoVisualizer>