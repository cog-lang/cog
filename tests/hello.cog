// hello.cog

import cog;

// A basic "hello world" program

func main(
	argc : Int,
	argv : Ptr<Ptr<Char> >) -> Int
{
	printf("Hello, world!\n");

	return 0;
}