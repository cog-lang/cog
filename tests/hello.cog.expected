result: 0
stdout: {
Hello, world!
}
stderr: {
}
